### PR TITLE
Add and update logging in gatekeeper. 

### PR DESCRIPTION
**Pull request by pawelkisielewicz**
I've made 'two levels' of logging. INFO for regular-positive way (like endpoint received message, positive validation..) and WARNING for 'negative' infos- like validation errors. 

After @bartoszbetka and @cameel  comments there is no more logs related invalid messages from client in warnings

### CODE REVIEW

File: concent_api/gatekeeper/utils.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 from django.http import JsonResponse
 
+from utils import logging
 
-def gatekeeper_access_denied_response(message, error_code = None, path = None, subtask_id = None, client_key = None):
+
+def gatekeeper_access_denied_response(message, method_origin, error_code=None, path=None, subtask_id=None,
```
Comment (kbeker) [inline concent_api/gatekeeper/utils.py]: If you want break lines better way will be to do something like that:
```python
def gatekeeper_access_denied_response(
    message,
    method_origin,
    error_code=None,
    path=None,
    subtask_id=None,
    client_key=None
):

File: concent_api/gatekeeper/utils.py [hunk 2]
```diff
@@ -10,6 +13,13 @@ def gatekeeper_access_denied_response(message, error_code = None, path = None, s
         'client_key':      client_key,
     }
 
+    logging.log_gatekeeper_access_denied_response_messages(method_origin,
```
Comment (kbeker) [inline concent_api/gatekeeper/utils.py]: First argument shouldn't be in the same line as function call

File: concent_api/gatekeeper/tests/test_utils.py [hunk 3]
```diff
@@ -19,6 +19,7 @@ def setUp(self):
     def test_gatekeeper_access_denied_response_should_return_appropriate_body_and_headers(self):
         response = gatekeeper_access_denied_response(
             self.message,
+            'Upload',
```
Comment (rwrzesien) [inline concent_api/gatekeeper/tests/test_utils.py]: Please use `Operation` enum value from golem_messages.

File: concent_api/gatekeeper/utils.py (hunk 2 above)
Comment (rwrzesien) [reply concent_api/gatekeeper/utils.py]: Plus please fix indentation below.

File: concent_api/gatekeeper/views.py [hunk 4]
```diff
@@ -25,32 +23,32 @@
 from gatekeeper.utils               import gatekeeper_access_denied_response
 from utils.constants                import ErrorCode
 from utils.helpers                  import get_current_utc_timestamp
+from utils                          import logging
```
Comment (rwrzesien) [inline concent_api/gatekeeper/views.py]: Import order should be:
```
from utils                              import logging
from utils.constants              import ErrorCode
from utils.helpers                 import get_current_utc_timestamp
```

File: concent_api/gatekeeper/views.py [hunk 5]
```diff
@@ -25,32 +23,32 @@
...
-logger = logging.getLogger(__name__)
-
 VALID_SHA1_HASH_REGEX = re.compile(r"^[a-fA-F\d]{40}$")
 
 
 @csrf_exempt
 @require_POST
 def upload(request):
-    logger.debug("Upload request received.")
+    logging.log_gatekeeper_endpoint_request_received(request, 'Upload')
```
Comment (rwrzesien) [inline concent_api/gatekeeper/views.py]: Same here, please use enum value.

File: concent_api/gatekeeper/views.py [hunk 6]
```diff
@@ -61,33 +59,32 @@ def upload(request):
...
         assert isinstance(response_or_file_info, JsonResponse)
-        logger.info(json.loads(response_or_file_info.content.decode())['message'])
         return response_or_file_info
-    logger.info('Request passed all download validations.')
-
     return JsonResponse({"message": "Request passed all download validations."}, status = 200)
 
 
-def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransferToken.FileInfo, JsonResponse]:
+def parse_headers(request: WSGIRequest, path_to_file: str, method_origin: str) -> Union[FileTransferToken.FileInfo, JsonResponse]:
```
Comment (rwrzesien) [inline concent_api/gatekeeper/views.py]: This should be called 'operation'.

File: concent_api/gatekeeper/views.py [hunk 7]
```diff
@@ -356,11 +378,16 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
     matching_files = [file for file in loaded_golem_message.files if path_to_file == file['path']]
 
     if len(matching_files) == 1:
+        logging.log_gatekeeper_message_successful_validated(loaded_golem_message.operation,
```
Comment (rwrzesien) [inline concent_api/gatekeeper/views.py]: Please fix indentation.

File: concent_api/utils/logging.py [hunk 8]
```diff
@@ -220,3 +219,45 @@ def get_subtask_id_for_logging(message):
     if not isinstance(subtask_id, str):
         subtask_id = ''
     return subtask_id
+
+
+def log_gatekeeper_endpoint_request_received(request, method_origin: str):
```
Comment (rwrzesien) [inline concent_api/utils/logging.py]: `method_origin` => `operation`

File: concent_api/gatekeeper/views.py [hunk 9]
```diff
@@ -356,11 +378,18 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
     matching_files = [file for file in loaded_golem_message.files if path_to_file == file['path']]
 
     if len(matching_files) == 1:
+        logging.log_gatekeeper_message_successful_validated(
```
Comment (dybi) [inline concent_api/gatekeeper/views.py]: Should be `successfully` ;)

File: concent_api/utils/logging.py [hunk 10]
```diff
@@ -220,3 +219,50 @@ def get_subtask_id_for_logging(message):
     if not isinstance(subtask_id, str):
         subtask_id = ''
     return subtask_id
+
+
+def log_gatekeeper_endpoint_request_received(request, operation: str):
```
Comment (dybi) [inline concent_api/utils/logging.py]: Why don't you use `replace_element_to_unavailable_instead_of_none` a decorator here?

Comment (dybi) [inline concent_api/utils/logging.py]: I think I would shorten this name to `log_request_received`

File: concent_api/utils/logging.py [hunk 11]
```diff
@@ -220,3 +219,50 @@ def get_subtask_id_for_logging(message):
...
+    path_to_file = request.META['PATH_INFO'] if 'PATH_INFO' in request.META.keys() else '-path to file UNAVAILABLE-'
+    logger.info("{} request received. File '{}'".format(
+        operation,
+        path_to_file
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_gatekeeper_message_under_validation(operation_type: str, file_path: str, subtask_id: bytes, public_key: bytes):
```
Comment (dybi) [inline concent_api/utils/logging.py]: I would shorten this to: `log_message_under_validation`

File: concent_api/utils/logging.py [hunk 12]
```diff
@@ -220,3 +219,50 @@ def get_subtask_id_for_logging(message):
...
+    logger.info("{} request will be validated. File '{}', with subtask_id '{}'. Client public key: '{}'".format(
```
Comment (dybi) [inline concent_api/utils/logging.py]: Please logg message type here as well

File: concent_api/utils/logging.py [hunk 13]
```diff
@@ -220,3 +219,50 @@ def get_subtask_id_for_logging(message):
...
+        operation_type.capitalize(),
+        file_path,
+        subtask_id,
+        public_key
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_gatekeeper_message_successful_validated(operation: str, file_path: str, subtask_id: bytes, public_key: bytes):
```
Comment (dybi) [inline concent_api/utils/logging.py]: Should be called `log_message_under_validation`, and message type should be logged as well

File: concent_api/utils/logging.py [hunk 14]
```diff
@@ -220,3 +219,50 @@ def get_subtask_id_for_logging(message):
...
+        file_path,
+        subtask_id,
+        public_key
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_gatekeeper_access_denied_response_messages(
+        upload_or_download: str,
+        msg: str,
```
Comment (dybi) [inline concent_api/utils/logging.py]: is it `message` or `error_code`?

File: concent_api/utils/logging.py [hunk 15]
```diff
@@ -220,3 +219,50 @@ def get_subtask_id_for_logging(message):
...
+    logger.info("{} request passed all validations. File '{}', with subtask_id '{}'. Client public key: '{}'".format(
+        operation.capitalize(),
+        file_path,
+        subtask_id,
+        public_key
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_gatekeeper_access_denied_response_messages(
```
Comment (dybi) [inline concent_api/utils/logging.py]: `log_access_denied`?

File: concent_api/utils/logging.py (hunk 10 above)
Comment (pawelkisielewicz) [reply concent_api/utils/logging.py]: Because I log two variables:
-  operation, it can't be none. This method is called 2 times and this value is always given. 
- path_to_file. I get it from request and handle it in this method. 
Use of this decorator would work, but I don't think it's necessary

File: concent_api/utils/logging.py (hunk 14 above)
Comment (pawelkisielewicz) [reply concent_api/utils/logging.py]: It's message. Error_code is just below

File: concent_api/utils/logging.py (hunk 15 above)
Comment (pawelkisielewicz) [reply concent_api/utils/logging.py]: This function logs  why access was denied. It's called in "def gatekeeper_access_denied_response()" in utils, so I called it like this. If you think I should change it i will

File: concent_api/gatekeeper/views.py (hunk 9 above)
Comment (pawelkisielewicz) [reply concent_api/gatekeeper/views.py]: done :)

File: concent_api/utils/logging.py (hunk 14 above)
Comment (rwrzesien) [reply concent_api/utils/logging.py]: Please use full names instead of shortcuts.

File: concent_api/utils/logging.py (hunk 15 above)
Comment (pawelkisielewicz) [reply concent_api/utils/logging.py]: I try to unify messages in all apps and use the same function for all of them. I don't know if it will be possible for all, but I try. Names of log methods are changed

File: concent_api/gatekeeper/utils.py [hunk 16]
```diff
@@ -1,16 +1,35 @@
 from django.http import JsonResponse
+from golem_messages.message import FileTransferToken
 
+from utils import logging
 
-def gatekeeper_access_denied_response(message, error_code = None, path = None, subtask_id = None, client_key = None):
+
+def gatekeeper_access_denied_response(
+        message: str,
```
Comment (rwrzesien) [inline concent_api/gatekeeper/utils.py]: Please fix indentation.

File: concent_api/utils/logging.py [hunk 17]
```diff
@@ -224,3 +224,63 @@ def get_subtask_id_for_logging(message):
...
+def log_request_received(path_to_file, operation: FileTransferToken.Operation):
+    logger.debug("{} request received. Path to file: '{}'".format(
+        operation.capitalize(),
+        path_to_file
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_message_under_validation(
+        operation: FileTransferToken.Operation,
```
Comment (rwrzesien) [inline concent_api/utils/logging.py]: Please fix indentation.

File: concent_api/utils/logging.py [hunk 18]
```diff
@@ -224,3 +224,63 @@ def get_subtask_id_for_logging(message):
...
+        message_type,
+        file_path,
+        subtask_id,
+        public_key
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_message_successfully_validated(
+        operation: FileTransferToken.Operation,
```
Comment (rwrzesien) [inline concent_api/utils/logging.py]: Please fix indentation.

File: concent_api/utils/logging.py [hunk 19]
```diff
@@ -224,3 +224,63 @@ def get_subtask_id_for_logging(message):
...
+        message_type,
+        file_path,
+        subtask_id,
+        public_key
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_message_validation_failed(
+        operation: FileTransferToken.Operation,
```
Comment (rwrzesien) [inline concent_api/utils/logging.py]: Please fix indentation.

File: concent_api/utils/logging.py [hunk 20]
```diff
@@ -224,3 +224,63 @@ def get_subtask_id_for_logging(message):
...
+    ))
+
+
+@replace_element_to_unavailable_instead_of_none
+def log_message_under_validation(
+        operation: FileTransferToken.Operation,
+        message_type: str,
+        file_path: str,
+        subtask_id: bytes,
+        public_key: bytes):
```
Comment (rwrzesien) [inline concent_api/utils/logging.py]: Please move `)` to next line.
